utils/trade.py

```diff
@@ -99,7 +99,7 @@ def get_item_modifiers_by_text() -> dict:
         return mod_list_dict_text
     else:
         item_modifiers = get_item_modifiers()
-        mod_list_dict_text = {item_modifiers[i]["id"]: i for i in range(len(item_modifiers))}
+        mod_list_dict_text = {(item_modifiers[i].text, item_modifiers[i].type): item_modifiers[i] for i in range(len(item_modifiers))}
```

Comment:
Awesome job. Can you please redo the lookup functions in parse.py so that we take advantage of this faster lookup utility? Seems easier to just have this PR rewrite the lookup than to break it into two parts.

Reply:
The algorithm used in find_affix_match to do the lookup on these is fundamentally different than performing a dict lookup, so it must be changed to conform with upstream in order for this patch to make any difference.

However, Df is already far into some work with reworking how we both parse and lookup items, which includes these lookups, so I'm not sure that I should change how this works in upstream at the moment.

I believe we should wait for Df's rewrites for this to be a thing.